lib/active_record_proxy_adapters/primary_replica_proxy.rb

```diff
@@ -46,7 +46,7 @@ class PrimaryReplicaProxy # rubocop:disable Metrics/ClassLength
     ].map(&:freeze).freeze
 
     # Abstract adapter methods that should be proxied.
-    hijack_method(*ActiveRecordContext.hijackable_methods)
+    hijack_method(*ActiveRecordContext.hijackable_methods, :exec_insert)
```

Comment:
@ankane can we move this change to `ActiveRecordProxyAdapters::TrilogyProxy` for now? I think this will need some refactoring but as a temporary fix, I'd like to minimize the impact on other adapters.

```ruby
  class TrilogyProxy < Mysql2Proxy
    hijack_method :exec_insert
  end
```